### PR TITLE
[RISCV] Add DAG combine for (vmv_s_x_vl (undef) (vmv_x_s X).

### PR DESCRIPTION
**Pull request by topperc**
We can use the original vector as long as the type of X matches the result type of the vmv_s_x_vl.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 1]
```diff
@@ -16791,6 +16791,11 @@ SDValue RISCVTargetLowering::PerformDAGCombine(SDNode *N,
     SDValue Scalar = N->getOperand(1);
     SDValue VL = N->getOperand(2);
 
+    if (Scalar.getOpcode() == RISCVISD::VMV_X_S &&
+        Scalar.getOperand(0).getValueType() == N->getValueType(0) &&
+        isa<ConstantSDNode>(N->getOperand(2)) && N->getOperand(0).isUndef())
```
Comment (lukel97) [inline llvm/lib/Target/RISCV/RISCVISelLowering.cpp:16796]: Why do we need to check VL is constant?

Comment (topperc) [reply llvm/lib/Target/RISCV/RISCVISelLowering.cpp]: We don't. I found this code in my downstream and didn't scrutinize it.

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 2]
```diff
@@ -16791,6 +16791,11 @@ SDValue RISCVTargetLowering::PerformDAGCombine(SDNode *N,
     SDValue Scalar = N->getOperand(1);
     SDValue VL = N->getOperand(2);
 
+    if (Scalar.getOpcode() == RISCVISD::VMV_X_S &&
+        Scalar.getOperand(0).getValueType() == N->getValueType(0) &&
+        N->getOperand(0).isUndef())
```
Comment (lukel97) [inline llvm/lib/Target/RISCV/RISCVISelLowering.cpp:16796]:
```suggestion
        Passthru.isUndef())
```

**Review by lukel97 — APPROVED** [review]

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Pull request merged**